harness/determined/keras/callbacks.py

```diff
@@ -701,3 +701,22 @@ def on_train_workload_begin(
     def on_train_workload_end(self, total_batches_trained: int, logs: Dict) -> None:
         tf.keras.callbacks.TensorBoard.on_epoch_end(self, self.workload_end_count, logs)
         self.workload_end_count += 1
+
+
+class _DeterminedProfiler(Callback):
+    """Hooks Keras into the profiler agent lifecycle"""
+
+    def __init__(self, prof: profiler.ProfilerAgent) -> None:
+        super().__init__()
+        self.current_batch = 0
+        self.prof = prof
+
+    def get_state(self):
+        return {"current_batch": self.current_batch}
+
+    def load_state(self, state: Any) -> None:
+        self.current_batch = state["current_batch"]
+
+    def on_train_batch_begin(self, batch: int, logs: Optional[Dict] = None) -> None:
```

Comment:
@shiyuann `logs` is part of the abstract class's contract here so mypy complains if I don't include it in the signature.

Reply:
I think what shiyuan was probably getting at was that you should call it `_: Optional[Dict] = None` to make it clear it's not being used.

Reply:
also fixed